lib/cli/cliCert.js

```diff
@@ -0,0 +1,139 @@
+        // security properties. It may be useful to set this flag to test certain features that you plan to run in
+        // environments that are different from your testing environment.
+        forge.options.usePureJavaScript = false;
+
+        const { callback, dbConnect } = this.options;
+        const id = "system.certificates";
+        const certPropPath = "native.certificates";
+
+        // https://github.com/digitalbazaar/forge
+        const pki = forge.pki;
+
+        const keys = pki.rsa.generateKeyPair({bits: 2048, e: 0x10001});
+        let cert = pki.createCertificate();
+
+        cert.publicKey = keys.publicKey;
+        cert.serialNumber = '01';
+        cert.validity.notBefore = new Date();
+        cert.validity.notAfter = new Date();
+        cert.validity.notAfter.setFullYear(cert.validity.notBefore.getFullYear() + 5);
+
+        const subAttrs = [
+            {name:'commonName',value: tools.getHostName()},
+            {name:'organizationName',value:'ioBroker GmbH'},
+            {shortName:'OU',value:'iobroker'}
+        ];
+
+        const issAttrs = [
+            {name:'commonName',value: 'iobroker'},
+            {name:'organizationName',value:'ioBroker GmbH'},
+            {shortName:'OU',value:'iobroker'}
+        ];
+
+        cert.setSubject(subAttrs);
+        cert.setIssuer(issAttrs);
+
+        cert.setExtensions([{
+            name: 'basicConstraints',
+            critical: true,
+            cA: false
+            },
+            {
+                name: 'keyUsage',
+                critical: true,
+                digitalSignature: true,
+                contentCommitment: true,
+                keyEncipherment: true,
+                dataEncipherment: true,
+                keyAgreement: true,
+                keyCertSign: true,
+                cRLSign: true,
+                encipherOnly: true,
+                decipherOnly: true
+            },
+            {
+                name: 'subjectAltName',
+                altNames: [{
+                    type: 2,
+                    value: os.hostname()
+                }]
+            },
+            {
+                name: 'subjectKeyIdentifier'
+            },
+            {
+                name: 'extKeyUsage',
+                serverAuth: true,
+                clientAuth: true,
+                codeSigning: false,
+                emailProtection: false,
+                timeStamping: false
+            },
+            {
+                name: 'authorityKeyIdentifier'
+            }
+        ]);
+
+        cert.sign(keys.privateKey);
+
+        const pem_pkey = pki.privateKeyToPem(keys.privateKey);
+        const pem_cert = pki.certificateToPem(cert);
+
+        console.log(pem_pkey);
+        console.log(pem_cert);
+
+        const certificates = {
+            defaultPrivate: pem_pkey,
+            defaultPublic: pem_cert
+        };
+
+        const command = ["set", id, certPropPath + "=" + JSON.stringify(certificates)];
```

Comment:
I would add a comment that this is more or less a shortcut for `iobroker object set ...`